get_http.go

```diff
@@ -293,50 +294,66 @@ func (g *HttpGetter) Get(dst string, u *url.URL) error {
 	// If there is a subdir component, then we download the root separately
 	// into a temporary directory, then copy over the proper subdir.
 	source, subDir := SourceDirSubdir(source)
-	if subDir == "" {
-		var opts []ClientOption
+	if subDir != "" {
+		// We have a subdir, time to jump some hoops
+		return g.getSubdir(ctx, dst, source, subDir)
```

Comment:
I was a little confused about the call to getSubdir the first time I read this code. Is the source being used here the value returned from the X-Terraform-Get header? 

If so, would we not want to run the protocol switch check before returning?  Or will this be handled in the final call to Get that is made in the getSubdir function?

Reply:
That's a good question. While the PR looks big because I dropped a level of indentation by returning early, I left the overall logic the same. I think you may be right that a subdir would bypass the the protocol checks since it goes on to call `Get` just like below -- I'll follow up with a refactor of `getSubdir` so we can pass in the same options

Reply:
The new proto checks were bypassed by the addition of a subdirectory component to the url. I refactored this a bit to move the `getSubdir` call after the checks, and pass in the same options we would use for `Get`.